crates/teloxide-core/src/types/message.rs

```diff
@@ -119,19 +120,20 @@ pub struct MessageCommon {
     /// `true`, if the message is sent to a forum topic.
     // FIXME: `is_topic_message` is included even in service messages, like ForumTopicCreated.
     //        more this to `Message`
-    #[serde(default)]
+    #[serde(default, skip_serializing_if = "std::ops::Not::not")]
     pub is_topic_message: bool,
 
     /// `true`, if the message is a channel post that was automatically
     /// forwarded to the connected discussion group.
-    #[serde(default)]
+    #[serde(default, skip_serializing_if = "std::ops::Not::not")]
     pub is_automatic_forward: bool,
 
     /// `true`, if the message can't be forwarded.
-    #[serde(default)]
+    #[serde(default, skip_serializing_if = "std::ops::Not::not")]
     pub has_protected_content: bool,
 }
 
+#[serde_with_macros::skip_serializing_none]
```

Comment:
I don't think adding `#[serde_with_macros::skip_serializing_none]` helps here, this structure doesn't have any `Option` fields, so such annotations do nothing, while still needing to process code.

Reply:
Discussed in PM. It's okay :D